java-api/src/main/java/com/sap/cloud/security/config/ClientIdentity.java

```diff
@@ -34,7 +36,8 @@ default boolean isValid() {
 	 * @return the boolean
 	 */
 	default boolean isCertificateBased() {
-		return hasValue(getCertificate()) && hasValue(getKey());
+		return hasValue(getCertificate()) && hasValue(
```

Comment:
the formatting looks weird, can we put the two different combinations on one line each maybe?